include/rocksdb/memtablerep.h

```diff
@@ -57,15 +60,28 @@ class MemTableRep {
   // concatenated with values.
   class KeyComparator {
    public:
+    KeyComparator(const InternalKeyComparator& cmp) : comparator_(cmp) {}
+
     // Compare a and b. Return a negative value if a is less than b, 0 if they
     // are equal, and a positive value if a is greater than b
-    virtual int operator()(const char* prefix_len_key1,
-                           const char* prefix_len_key2) const = 0;
-
-    virtual int operator()(const char* prefix_len_key,
-                           const Slice& key) const = 0;
-
-    virtual ~KeyComparator() { }
+    int operator()(const char* prefix_len_key1,
+                   const char* prefix_len_key2) const {
+      // Internal keys are encoded as length-prefixed strings.
+      Slice k1 = GetLengthPrefixedSlice(prefix_len_key1);
+      Slice k2 = GetLengthPrefixedSlice(prefix_len_key2);
+      return comparator_.Compare(k1, k2);
```

Comment:
Same here and below. Moving to virtual doesn't mean that we have to declare the function here. If we declare it under .cc, I'm sure they will be inlined under -O3 too.